erpnext/controllers/buying_controller.py

```diff
@@ -491,6 +492,11 @@ def set_qty_as_per_stock_uom(self):
 					# Set Received Qty in Stock UOM
 					d.received_stock_qty = flt(d.received_qty) * flt(d.conversion_factor, d.precision("conversion_factor"))
 
+	def set_rate_of_stock_uom(self):
+		if self.doctype in ["Purchase Receipt", "Purchase Invoice", "Purchase Order"]:
+			for d in self.get("items"):
+				d.stock_uom_rate = d.rate / d.conversion_factor
```

Comment:
So the field `stock_uom_rate` is takes company currency (INR). `rate` field takes the currency in the transaction , lets say USD.
You are then dividing price in USD by conversion factor and displaying it in INR
![Screenshot 2021-01-19 at 8 49 36 PM](https://user-images.githubusercontent.com/25857446/105054434-00360080-5a98-11eb-8a95-110d3168e467.png)
`Rate (Stock UOM)` should have been **1 USD or 73.46 INR**
1 USD would be better, lets show rate in stock uom in transaction currency